### PR TITLE
[Merged by Bors] - chore: bump to nightly-2022-02-06

### PR DESCRIPTION
**Pull request by digama0**

### CODE REVIEW

File: Mathlib/Data/List/Defs.lean [hunk 1]
```diff
@@ -410,16 +410,16 @@ def rotate' : List α → ℕ → List α
 | a :: l, n+1 => rotate' (l ++ [a]) n
 
 def mmap {m : Type u → Type v} [Monad m] {α β} (f : α → m β) : List α → m (List β)
-| [] => []
-| h :: t => do (← f h) :: (← mmap f t)
+| [] => pure []
+| h :: t => do pure $ (← f h) :: (← mmap f t)
```
Comment (gebner) [inline Mathlib/Data/List/Defs.lean:414]:
```suggestion
| h :: t => return (← f h) :: (← mmap f t)
```

File: Mathlib/Data/List/Defs.lean [hunk 2]
```diff
@@ -435,8 +435,8 @@ Example: suppose `l = [1, 2, 3]`. `mmapUpperTriangle f l` will produce the list
 `[f 1 1, f 1 2, f 1 3, f 2 2, f 2 3, f 3 3]`.
 -/
 def mmapUpperTriangle {m} [Monad m] {α β : Type u} (f : α → α → m β) : List α → m (List β)
-| [] => []
-| h :: t => do (← f h h) :: (← t.mmap (f h)) ++ (← t.mmapUpperTriangle f)
+| [] => pure []
+| h :: t => do pure $ (← f h h) :: (← t.mmap (f h)) ++ (← t.mmapUpperTriangle f)
```
Comment (gebner) [inline Mathlib/Data/List/Defs.lean:439]:
```suggestion
| h :: t => return (← f h h) :: (← t.mmap (f h)) ++ (← t.mmapUpperTriangle f)
```

File: Mathlib/Tactic/Find.lean [hunk 3]
```diff
@@ -43,9 +43,9 @@ private partial def matchHyps : List Expr → List Expr → List Expr → MetaM
 -- from Lean.Server.Completion
 private def isBlackListed (declName : Name) : MetaM Bool := do
   let env ← getEnv
-  declName.isInternal
-  <||> isAuxRecursor env declName
-  <||> isNoConfusion env declName
+  pure $ declName.isInternal
+   || isAuxRecursor env declName
+   || isNoConfusion env declName
   <||> isRec declName
```
Comment (gebner) [inline Mathlib/Tactic/Find.lean:49]: It surprised me that this worked without parenthesizes.

**Comment by gebner** [comment]
bors merge

File: Mathlib/Tactic/Lint/Basic.lean [hunk 4]
```diff
@@ -34,14 +34,14 @@ initialize nolintAttr : ParametricAttribute (Array Name) ←
...
-        | `(attr|nolint $[$ids]*) => ids.mapM (·.getId.eraseMacroScopes)
+        | `(attr|nolint $[$ids]*) => pure $ ids.map (·.getId.eraseMacroScopes)
         | _ => throwError "unexpected nolint syntax {stx}"
   }
 
 /-- Returns true if `decl` should be checked
 using `linter`, i.e., if there is no `nolint` attribute. -/
 def shouldBeLinted [Monad m] [MonadEnv m] (linter : Name) (decl : Name) : m Bool := do
-  !((← nolintAttr.getParam (← getEnv) decl).getD {}).contains linter
+  pure !((nolintAttr.getParam (← getEnv) decl).getD {}).contains linter
```
Comment (digama0) [inline Mathlib/Tactic/Lint/Basic.lean:44]: @gebner Also check out the funky precedence here. `!` has precedence 40 on the right and `max` on the left, so it acts like `$` here.

File: Mathlib/Tactic/Find.lean (hunk 3 above)
Comment (digama0) [reply Mathlib/Tactic/Find.lean:49]: Yeah, the precedence is not what I would have expected. See also the `!` precedence below.

File: Mathlib/Tactic/Lint/Basic.lean (hunk 4 above)
Comment (gebner) [reply Mathlib/Tactic/Lint/Basic.lean:44]: omg

**Pull request closed without merging**